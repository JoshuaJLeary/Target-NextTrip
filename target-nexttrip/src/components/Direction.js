import React, { Component } from 'react';
import { connect } from 'react-redux';

class Direction extends Component {

  handleDirectionChange = (event) => {
    const { onChange, dispatch } = this.props;
    onChange({ direction: event.target.value });
    setTimeout( () => {
      dispatch({ type: 'GET_ROUTE_STOP', payload: this.props.nextTripInput.nextTrip })
    }, 1000);
  }

  render() {

    let routeDirection = this.props.nextTripRouteDirection.map(( direction ) => {
      console.log('this.props:', this.props);
      return (
        <option key={direction.Value} value={direction.Value}>
          {direction.Text}
        </option>
      )
    })

    return (
      <div className='inputSelectContainer'>
        <select className='inputSelect' onChange={this.handleDirectionChange}>
          <option>Select Direction</option>
          {routeDirection}
        </select>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  state
})

export default connect(mapStateToProps)(Direction);
