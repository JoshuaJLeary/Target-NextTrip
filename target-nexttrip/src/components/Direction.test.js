import React from 'react';
import ReactDOM from 'react-dom';
import Direction from './Direction';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

import { createStore, applyMiddleware } from 'redux';
import reducer from '../reducers/reduxStore';
import { Provider } from 'react-redux';
import createSagaMiddleware from 'redux-saga';

const mockDirection = [
  {
    Text: 'EASTBOUND',
    Value: '2'
  }
];

const sagaMiddleware = createSagaMiddleware();

const store = createStore(
  reducer,
  (applyMiddleware(sagaMiddleware)),
);

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <Direction nextTripRouteDirection={mockDirection}/>
    </Provider>, div);
});

it('renders Direction correctly', () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <Direction nextTripRouteDirection={mockDirection}/>
    </Provider>);
  expect(getByTestId('direction')).toHaveTextContent('EASTBOUND');
 });

 it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Direction nextTripRouteDirection={mockDirection}/>
    </Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
