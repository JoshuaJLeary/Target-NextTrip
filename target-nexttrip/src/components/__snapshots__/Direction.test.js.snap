// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="inputSelectContainer"
  data-testid="direction"
>
  <select
    className="inputSelect"
    onChange={[Function]}
  >
    <option>
      Select Direction
    </option>
    <option
      value="2"
    >
      EASTBOUND
    </option>
  </select>
</div>
`;
